ports/esp32/modmachine.c

```diff
@@ -134,7 +134,26 @@ STATIC mp_obj_t machine_sleep_helper(wake_type_t wake_type, size_t n_args, const
         esp_sleep_enable_timer_wakeup(((uint64_t)expiry) * 1000);
     }
 
-    #if !CONFIG_IDF_TARGET_ESP32C3
+    #if CONFIG_IDF_TARGET_ESP32C3
+
+    if (machine_rtc_config.ext1_pins != 0) {
+        for (int i = 0; i < GPIO_NUM_MAX; ++i) {
+            uint64_t bm = 1ULL << i;
+
+            if (machine_rtc_config.ext1_pins & bm) {
+                if (!esp_sleep_is_valid_wakeup_gpio((gpio_num_t)i)) {
+                    mp_raise_ValueError(MP_ERROR_TEXT("invalid wake-up port"));
+                }
+                gpio_sleep_set_direction((gpio_num_t)i, GPIO_MODE_INPUT);
+            }
+        }
```

Comment:
The problem here is that this deffinition may differ for use of light and deep sleep (as you stated above that for light sleep there can be used also non-RTC pins). If we would like to have pre-check, then we have to modify interface and define e.g. `esp32.wake_gpio_lightsleep` and `esp32.wake_gpio_deepsleep` instead of one shared `esp32.wake_ext1`. Or add optional argument to `esp32.deepsleep`/`esp32.lightsleep`, who knows :wink:. But this is more question for someone responsible for esp module architecture.